### PR TITLE
DSD-45: Documentation updates to the readme, pull request template, and the contributing files

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes [DSD-45](https://jira.nypl.org/browse/DSD-45)

## **This PR does the following:**
- Adds a `contributing` file and updates the `readme` and `pull request template` files to better outline and make it clear to a developer what they should do to contribute.
- We also have a wiki with useful and necessary information that we weren't linking to. This adds links to important information in relevant sections. 
- Also cleaning up existing documentation that is either no longer necessary or has changed in the past few weeks.

### Front End Review:
- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
Also went through the pages in the wiki and updated them where needed.

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+Next, make sure to read the [Contributing to the React Library](https://github.com/NYPL/nypl-design-system/wiki/Contributing-to-the-React-Library) wiki page for an in-depth explanation on creating meaningful updates that follow our engineering conventions.
+
+## React Component Guidelines
+
+There are specific guidelines we MUST follow when creating components in order to maintain engineering principles and conventions that also expand into the wider NYPL Digital department.
+
+### Components
+
+When creating a new React component or updating an existing one, you MUST follow the [Anatomy of a Component](https://github.com/NYPL/nypl-design-system/wiki/Anatomy-of-a-Component) wiki page documentation. This explains the all the files that make up a React component though it's possible to create or update a component without needing to create or edit a file mentioned. For example, if there is a styling error in the `Checkbox` component, it's possible you do not need to update its relevant test file.
```
Comment (bigfishdesign13) [inline .github/CONTRIBUTING.md]: Typo in second sentense.
> This explains `the` all the files that...

File: README.md [hunk 2]
```diff
@@ -59,44 +100,82 @@ function NewComponent(props) {
...
+<link href="https://unpkg.com/@nypl/design-system-react-components@0.9.1/dist/styles.css">
+```
 
-You can check out a working Codepen with unpkg [here](https://codepen.io/scoooooooby/pen/wvGzebO).
+You can check out a working Codepen with unpkg [here](https://codepen.io/edwinguzman/pen/ExmXGKx).
 
-## Design System and Product Requirements
+## Accessibility Product Requirements
 
 The NYPL Design System provides accessible stories, but real live data can necessitate additional accessibility requirements beyond what we're committed to in our generic, extensible components. To ensure your products' final result is accessible, please adhere to the accessibility requirements put together by NYPL's accessibility coordinator on [Metronome](http://themetronome.co/).
```
Comment (bigfishdesign13) [inline README.md:158]: Metronome is now password protected.  Maybe add something like, "For access to Metronome, please contact..."

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
These updates are awesome.  I left just a few comments about small things.

**Comment by hokei** [comment]
this looks great to me!

**Comment by EdwinGuzman** [comment]
Updated.

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.

**Pull request merged**